rmf_tasks/include/rmf_tasks/agv/State.hpp

```diff
@@ -24,45 +24,67 @@
 #include <rmf_utils/optional.hpp>
 
 #include <rmf_traffic/Time.hpp>
+#include <rmf_traffic/agv/Planner.hpp>
 
 namespace rmf_tasks {
 namespace agv {
 
+/// This state representation is used for task planning.
 class State
 {
 public:
 
   /// Constructor
   ///
-  /// \param[in] waypoint
+  /// \param[in] plan_start
+  ///   Current state's next start for new plans, includes the time which
+  ///   the plan can feasibly start, according to the finishing time of any
+  ///   tasks that the robot is currently performing.
+  ///
   /// \param[in] charging_waypoint
-  /// \param[in] finish_time
+  ///   The charging waypoint index of this robot.
+  ///
   /// \param[in] battery_soc
-  /// \param[in] threshold_soc
+  ///   Current battery state of charge of the robot. This value needs to be
+  ///   between 0.0 to 1.0.
   State(
-    std::size_t waypoint, 
+    rmf_traffic::agv::Plan::Start plan_start,
```

Comment:
I think the name `plan_start` is misleading and should be re-named to `location` as it is meant to reflect the expected finish location of the robot. This would align with the usage in `RobotContext` and `TaskManager` https://github.com/osrf/rmf_core/blob/89e758ac9b4fddc0302dac2d10d062ad7fadb477/rmf_fleet_adapter/src/rmf_fleet_adapter/TaskManager.hpp#L61

Reply:
Hmm, I was using `location` previously but found the name and the type not complementing each other, as it is not immediately clear what `location` has to do with `Plan::Start`.

Reply:
For now I will prioritize how semantic the names are, we can decide down the road if it really becomes confusing :man_shrugging: